src/integrationTest/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersAuditTest.java

```diff
@@ -0,0 +1,147 @@
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis"})
+public class CasePaymentOrdersAuditTest extends BaseTest {
+
+    @Autowired
+    private CasePaymentOrdersRepository casePaymentOrdersRepository;
+
+    @Autowired
+    private CasePaymentOrderMapper casePaymentOrderMapper;
+
+    @Autowired
+    EntityManagerFactory entityManagerFactory;
+
+    private CasePaymentOrderEntity casePaymentOrderEntity;
+
+    private AuditReader auditReader;
+
+    private static final String ACTION_VALUE = "Action";
+    private static final String NEW_ACTION_VALUE = "New Action";
+    private static final boolean SELECT_DELETED_ENTITIES = true;
+    private static final boolean SELECT_ENTITIES_ONLY = true;
+    private static final boolean SELECT_REVISION_INFO_ONLY = false;
+
+    @BeforeEach
+    void setUp() {
+        CasePaymentOrderEntity casePaymentOrderEntityToBePersisted =
+                casePaymentOrderMapper.toEntity(CasePaymentOrder.builder()
+                .orderReference(RandomStringUtils.random(10, true, true))
+                .action(ACTION_VALUE)
+                .caseId(new Random().nextLong())
+                .createdBy("gsullivan")
+                .effectiveFrom(LocalDateTime.now())
+                .responsibleParty("mrresponsible")
+                .build()
+        );
+        casePaymentOrderEntity = casePaymentOrdersRepository.save(casePaymentOrderEntityToBePersisted);
+        assertNotNull(casePaymentOrderEntity);
+
+        auditReader = AuditReaderFactory.get(entityManagerFactory.createEntityManager());
+    }
+
+    @Test
+    void testAuditOfCreatingCasePaymentOrder() {
+        CasePaymentOrderEntity auditedCasePaymentOrderEntity =
+                (CasePaymentOrderEntity) getAuditQuery(SELECT_ENTITIES_ONLY).getSingleResult();
+
+        assertNotNull(auditedCasePaymentOrderEntity);
+
+        assertEquals(casePaymentOrderEntity.getId(), auditedCasePaymentOrderEntity.getId());
+
+        assertAuditDataRevisionTypes(List.of(RevisionType.ADD));
+    }
+
+    @Test
+    void testAuditOfUpdatingCasePaymentOrder() {
+        casePaymentOrderEntity.setAction(NEW_ACTION_VALUE);
+
+        CasePaymentOrderEntity updateActionEntity = casePaymentOrdersRepository.save(casePaymentOrderEntity);
+
+        assertNotNull(updateActionEntity);
+
+        List auditedEntries = getAuditQuery(SELECT_ENTITIES_ONLY).getResultList();
+
+        // expect two audit entries - one for the create and one for the update
+        assertEquals(2, auditedEntries.size());
+
+        // check original value and updated value
+        assertEquals(ACTION_VALUE, ((CasePaymentOrderEntity)auditedEntries.get(0)).getAction());
+        assertEquals(NEW_ACTION_VALUE, ((CasePaymentOrderEntity)auditedEntries.get(1)).getAction());
+
+        assertAuditDataRevisionTypes(List.of(RevisionType.ADD, RevisionType.MOD));
+    }
+
+    @Test
+    void testAuditOfCreatingAndDeletingCasePaymentOrder() {
```

Comment:
```suggestion
    void testAuditOfDeletingCasePaymentOrder() {
```
this name is not consistent with the others. We are not testing the creation right? It's been tested in the dedicated test method already

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/29/commits/448f8f49d429b7c4e7617d980aef2ce4bc8e100e